gpflow/kernels/stationaries.py

```diff
@@ -171,17 +211,18 @@ def K_r(self, r):
         return self.variance * (1.0 + sqrt5 * r + 5.0 / 3.0 * tf.square(r)) * tf.exp(-sqrt5 * r)
 
 
-class Cosine(Stationary):
+class Cosine(AnisotropicStationary):
     """
     The Cosine kernel. Functions drawn from a GP with this kernel are sinusoids
     (with a random phase).  The kernel equation is
 
-        k(r) = σ² cos{r}
+        k(r) = σ² cos{2πd}
 
     where:
-    r  is the Euclidean distance between the input points, scaled by the lengthscale parameter ℓ,
+    d  is the sum of the per-dimension differences between the input points, scaled by the lengthscale parameter ℓ (i.e. Σ_i [(X - X2ᵀ) / ℓ]_i),
     σ² is the variance parameter.
     """
 
-    def K_r(self, r):
-        return self.variance * tf.cos(r)
+    def K_d(self, d):
+        d = tf.reduce_sum(d, axis=-1)
+        return self.variance * tf.cos(2 * np.pi * d)
```

Comment:
You're changing the behaviour of Cosine here. Is this intentional? If so, could you make this change and your motivation clear in the PR description please (as it'll be breaking backwards compatibility)?

Reply:
I'm not. In the previous version of this pull request, K_d would get the already summed matrix. Since we're now making an extra Anisotropic subclass, one could imagin that other kernels need to do something different with the distance, so now the scaled_difference_matrix returns a matrix of distance vectors, and it's up to the kernel to to whatever it wants with it.

Reply:
I meant the factor 2π between introduced by this change, not the sum!

Reply:
Right, sorry. Yes, this is intentional, since this makes the lenghtscale parameter directly interpretable as period length. Given that gpflow 2 is, as far as I can tell, anyway not backwards compatible to gpflow 1 (i.e. models from 1 can not be loaded into 2) and that the Cosine kernel seems to not be widely used (otherwise someone would have noticed that it doesn't work with multivariate data much sooner), I don't think that this is a big issue.